src/components/api/json-rpc/che-json-rpc-master-api.ts

```diff
@@ -126,9 +130,12 @@ export class CheJsonRpcMasterApi {
   }
 
   reconnect(entrypoint: string): void {
-    this.$log.warn('WebSocket connection is closed.');
-    if (this.reconnectionAttemptNumber === this.maxReconnectionAttempts) {
-      this.$log.warn('The maximum number of attempts to reconnect WebSocket has been reached.');
+    if (this.reconnectionAttemptNumber < this.maxReconnectionAttempts) {
+      this.$log.warn('WebSocket connection is closed.');
+    } else {
+      const warningMessage = 'The maximum number of attempts to reconnect WebSocket has been reached. WebSocket connection is failed.';
```

Comment:
```suggestion
      const warningMessage = 'Reached the limit of tries to establish WebSocket connection. Check browser logs to figure out the cause which may be Che Server unavailability or network issues';
```

Reply:
It would be ideal to add a reference to documentation to troubleshoot that, but I'm not aware of such.

@l0rd since you're created of the issue - any suggestions about that message?

Reply:
I would say that "Websockets connections are failing due to network restrictions. Che workspaces may not be usable. Please refer to the Network Troubleshooting section of the Che User Guide". 

Then you should add a Network Troubleshooting section that provides more hints. From the email that I have forwarded you:

>  - Provide useful suggestions to the end user - now this might be tricky, since “Switch off your VPN” or “Disable local firewall" is never going to be reasonable option in a corporate deployment however suggesting this is a network, firewall, VPN or proxy issue might be a useful clue about who to report this problem to.
>   - Some suggested links on how to configure VPN, firewalls, proxies etc. These maybe third party, but we should be trying to collect some resolutions stories that we can share as a guide to the network team particularly if they are not familiar with wss. If we can also find some security articles that are positive about websockets they would also be useful to share.
>  - It would be really useful to run or recommend a 3rd party web site that has wss content that can be used as a test site with no login wall. So if this site works, then CRW should be fine. That allows any network engineer to have something to resolve against without needing access to CRW ( https://websocketstest.com/)
>   - Document the network requirement in the CRW Installation/Admin guide so there is at least a section highlighting this.

Reply:
cc @alexgroom @jamesfalkner

Reply:
@l0rd Thx. There is a dedicated issue to create Trouble Shooting documentation about that https://github.com/eclipse/che/issues/17468

Reply:
from Rick https://github.com/eclipse/che/issues/17468#issuecomment-664551544

> In writing the error messages, could we please be specific about which websocket endpoint is having trouble connecting to which other endpoint? (Identify the targeted component in easy to recognize terms.)
The reason for this request: sometimes environments are not homogenous (i.e. on AWS different Che containers may end up in different 'Availability Zones'). Things like that can make network debugging mysterious, but if we know from the error message exactly where we are trying to get to, it can be a good head start.

Reply:
I have done